sw/host/opentitanlib/src/crypto/spxplus.rs

```diff
@@ -0,0 +1,164 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+use std::io::{Read, Write};
+
+use anyhow::Result;
+
+use pqcrypto_sphincsplus::sphincsshake256128ssimple as spx;
+use pqcrypto_traits::sign::DetachedSignature;
+use pqcrypto_traits::sign::PublicKey;
+use pqcrypto_traits::sign::SecretKey;
+
+use crate::util::bigint::fixed_size_bigint;
+use crate::util::file::{FromReader, ToWriter};
+
+// Signature and key sizes taken from Table 8 on page 57 of the SPHINCS+ Round 3 Specification:
+// https://sphincs.org/data/sphincs+-round3-specification.pdf
+const PUBLIC_KEY_BYTE_LEN: usize = 32;
+const SECRET_KEY_BYTE_LEN: usize = 64;
+const SIGNATURE_BIT_LEN: usize = 7856 * 8;
+fixed_size_bigint!(Signature, at_most SIGNATURE_BIT_LEN);
+
+/// Trait for implementing public key operations.
+pub trait SpxPublicKeyPart {
+    /// Returns the public key component.
+    fn pk(&self) -> &spx::PublicKey;
+
+    fn pk_as_bytes(&self) -> &[u8] {
+        self.pk().as_bytes()
+    }
+
+    fn pk_len(&self) -> usize {
+        self.pk_as_bytes().len()
+    }
+
+    /// Verify a message signature, returning Ok(()) if the signature matches.
+    fn verify(&self, message: &[u8], sig: &SpxSignature) -> Result<()> {
+        spx::verify_detached_signature(
```

Comment:
@jon-flatley FYI: _detached_ in this context means signatures that don't include the message. This is basically equivalent to (in form, of course) the RSA signature scheme with appendix that we are already using. @jadephilipoom please correct me if I'm wrong.